rust-tests/cbmc-reg/Cast/powi.rs

```diff
@@ -0,0 +1,6 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+fn main() {
+    let _x: f64 = 1.0f64.powi(2);
```

Comment:
Add some sort of functional assert here.

Reply:
I can't assert anything meaningful since `powi` just returns `__nondet()`. Have confirmed that it doesn't get optimized out.

Reply:
If it returns `__nondet` then it is having the same behavior as any other non-defined function in CBMC. We should make sure the corresponding CBMC builtin is being used and returns something we can reason about.

Reply:
I haven't seen where the definitions actually are, but running the following in CBMC causes a verification failure:

```
#include <math.h>

int main() {
    assert(powi(1.0, 2) == powi(1.0, 2));
}
```

Reply:
Then it is not defined for sure. Thanks for looking into it!